api/v1beta1/cluster_types.go

```diff
@@ -149,7 +146,7 @@ type WorkersTopology struct {
 // MachineDeploymentTopology specifies the different parameters for a set of worker nodes in the topology.
 // This set of nodes is managed by a MachineDeployment object whose lifecycle is managed by the Cluster controller.
 type MachineDeploymentTopology struct {
```

Comment:
WDYT about setting same labels/annotations in our e2e testing? (clusterclass-quick-start.yaml & cluster-with-topology.yaml)
Just to get some easy happy path e2e test coverage.

I wouldn't introduce any additional code to our tests, but just having this in our e2e tests without validation is already a benefit (as we can e.g. verify by looking at the test artifacts).